sdk-setup/src/mb2

```diff
@@ -1808,6 +1812,48 @@ git_commit_brief() {
     return 0
 }
 
+create_patchset_id() {
+    local patchset_id=
+    if [[ ! $(git_ status --porcelain --untracked-files=no "$OPT_PKGDIR") ]]; then
+        patchset_id=$(git_ rev-parse HEAD)
+    else
+        patchset_id=$(git_ update-index --refresh >/dev/null; git_ stash create) \
+            || notice "Cannot save state of Git working tree: git-stash failed."
+    fi
+    : ${patchset_id:=$NULL_ID}
+
+    printf '%s\n' "$patchset_id"
+}
+
+recent_patches() {
+    local prune=$([[ $1 == --prune ]] && echo 1)
+
+    if [[ $prune ]]; then
+        git_ notes --ref "$PATCHSET_ID_REF" prune
+        git_ submodule foreach --quiet --recursive "git notes --ref '$PATCHSET_ID_REF' prune"
+    fi
+
+    local reasonable_limit=20
```

Comment:
Why 20? To me it seems recent_patches() is only used for determining if there are any recent patches, so one should be enough?

Reply:
This is used to detect whether patches have been applied, in which case it refuses to prepare again, so it needs to look behind possible commits created since that time.